errdefs/helpers_test.go

```diff
@@ -22,7 +22,7 @@ func TestNotFound(t *testing.T) {
 	if !cerrdefs.IsNotFound(e) {
 		t.Fatalf("expected not found error, got: %T", e)
 	}
-	if cause := e.(wrapped).Unwrap(); cause != errTest {
+	if cause := e.(wrapped).Unwrap(); cause != errTest { //nolint:errorlint
```

Comment:
🙈 The `//nolint:errorlint` that was added will very likely fail when you come to work on the `whyNoLint` rule for go-critic in https://github.com/moby/moby/pull/50171

Probably best to fix it straight-away, otherwise we end up updating these lines again. Could be something like;

```suggestion
	if cause := e.(wrapped).Unwrap(); cause != errTest { //nolint:errorlint // not using errors.Is, as this tests for the unwrapped error.
```